client/http/option.go

```diff
@@ -33,3 +35,14 @@ func CircuitBreaker(name string, set circuitbreaker.Setting) OptionFunc {
 		return nil
 	}
 }
+
+// Transport option for setting the Transport for the client.
+func Transport(rt http.RoundTripper) OptionFunc {
+	return func(tc *TracedClient) error {
+		if rt == nil {
+			return errors.New("transport must be supplied")
+		}
+		tc.cl.Transport = &nethttp.Transport{RoundTripper: rt}
```

Comment:
Question: Since the trace client has already integrated tracing, is there any need to use `nethttp` transport instead of `http` one?

Reply:
With a standard transport it panics [here](https://github.com/opentracing/opentracing-go/blob/master/ext/tags.go#L141), because of a nil span I think.

We could add logic to `TracedClient` to handle this but I thought it's more in the 'spirit' of `TracedClient` to retain all the tracing support without special cases. Also, the [nethttp.Transport](https://godoc.org/github.com/opentracing-contrib/go-stdlib/nethttp#Transport) is meant to wrap a RoundTripper so it's easy to inject it this way.

Reply:
Btw the codecov report seems wrong..

Reply:
Will it start a new span??? We are starting already one and a second one would be probably bad

Reply:
TracedClient was already using `nethttp.Transport`, set in the [constructor](https://github.com/beatlabs/patron/blob/master/client/http/http.go#L38), so there should be no change in behaviour. 


In this new `OptionFunc` we are simply setting the transport to be a `nethttp.Transport` with a specific inner transport.
If we don't do this then the inner transport defaults to `http.DefaultTransport` as you can see in the comment [here](https://github.com/opentracing-contrib/go-stdlib/blob/master/nethttp/client.go#L28).

Reply:
you are correct!